### ISSUE TITLE
Delve defined using go_repo cannot be used as DelveTool

### ISSUE THREAD

**Issue by marcuscaisey** [please-build/go-rules#182]
If you define delve as a third party dependency using a `go_repo` instead of `go_module` then `plz debug` fails. 

I think this is because with `go_repo`, a `go_binary` target is generated for the delve binary whereas the `go_module` rule generates a `build_rule` which does the compilation and bypasses `go_binary`. Both `go_binary` and `go_test` adds the delve tool to resulting `build_rule`'s `debug_tools` when building with `"dbg"` which means that the delve binary now gets built with a dependency on itself. 

I'm thinking we could either:
  1. Pass a flag to `go_binary` and `go_test` which disables the debug dependencies being added. The `please_go` `BUILD` file generator could then pass this flag when generating the `BUILD` files in the `go_repo` subrepos.
  2. Modify the `_debug_cmd` helper to not return anything for the configured `DELVE_TOOL`.

1. seems cleanest. It means that you couldn't run `plz debug` on a third party dependency but this just feels fine.

### Setup
Clone my minimal reproduction repo: https://github.com/marcuscaisey/plz-debug-go-bug.

```
plz debug //foo:foo_test
```

### Expected Behaviour
Delve spins up.
```
Type 'help' for list of commands.
(dlv)
```

### Actual Behaviour
`plz` exits with an error.
```
Building [0/1, 0.0s]:
  CPU use:   0.0%  I/O:   0.0%  Mem use:   0.0%
11:47:27.211 CRITICAL: Attempted to add ///third_party/go/github.com_go-delve_delve//cmd/dlv:dlv as a dependency of itself.
```
